Mathlib/Data/Set/Basic.lean

```diff
@@ -1575,6 +1575,10 @@ lemma disjoint_sdiff_left : Disjoint (t \ s) s := disjoint_sdiff_self_left
 lemma disjoint_sdiff_right : Disjoint s (t \ s) := disjoint_sdiff_self_right
 #align set.disjoint_sdiff_right Set.disjoint_sdiff_right
 
+theorem disjoint_sdiff_inter {α : Type*} (s t : Set α) : Disjoint (s \ t) (s ∩ t) :=
+  disjoint_of_subset_right (inter_subset_right _ _) disjoint_sdiff_left
```

Comment:
Ideally this would be proved in terms of a lattice lemma; and if that doesn't exist, we should write that lattice lemma first.

Reply:
Added a TODO